### DISCUSSION TITLE
BlockChain, IOT, Etherium, BTC , CryptoCurrency, Contratos inteligentes ...

### DISCUSSION THREAD

**Discussion by fredericksilva** [frontendbr/forum#624] [Discussão]
Devs do meu <3.
Estou montando uma listinha marota com desenvolvedores com proficiência nesses assuntos e pretendo disponibilizada de forma opensource <3 e quem sabe montar um repo de projetos opensource com blockchain  Brazilian Level!

Quem aqui esta nessa vibe?

Best Regards

**Comment by suissa** [comment]
Eu ainda tenho que pegar BlockChain e Ether(irum) esse ano. De IOT nao manjo nada.

**Comment by aferreira44** [comment]
Sou desenvolvedor .NET em uma das exchanges de Bitcoin aqui no Brasil.

Atualmente, estou estudando Node.JS e me aprofundando no Bitcoin Core e
Ethereum para desenvolver alguns projetos open-source e também lançar minha
própria exchange.

Tamo junto!

**Reply by Panzeinbok** [reply]
Levo muita fé na sua trajetória

**Comment by kazzkiq** [comment]
Eu to estudando blockchain e uns meses atrás criei um PWA pra acompanhar o preço das criptomoedas: https://cryptoholder.com/

É uma área muito promissora, e a quantidade de grana que tá rolando nos projetos **é absurda**. Tem muita vaga pra designers e devs (front-end incluso). Pra quem não se preocupa muito com estabilidade a médio e longo prazo, é uma boa oportunidade pra ganhar uns trocados trabalhando remoto.

**Comment by marciomansur** [comment]
Olá! Me interessei pelo grupo BR para trabalhar com blockchain.

Trabalho em uma empresa chamada ConsenSys. Desenvolvemos aplicações que utilizam smart contracts e Ethereum, até desenvolvi um pequeno boilerplate para quem quer iniciar o desenvolvimento de Dapps: https://github.com/marciomansur/simple-dapp-demo

Aqui tem uma imagem de docker para rodar um ambiente local para desenvolvimento de dapps (bem simples, usando testRPC e nodejs): https://github.com/marciomansur/docker-smart-contract-tests

Hoje estou mais na parte arquitetural das aplicações, focando mais em DevOps, porém, ainda gosto muito de acompanhar as novidades e estudar como o blockchain e a EVM funcionam. 

A tecnologia blockchain é muito promissora e em breve, muitas empresas brasileiras irão utilizar a tecnologia, acho que é sim uma boa ideia criar uma ponte de comunicação e ensinar as pessoas a utilizarem e criarem smart contracts e aplicações distribuídas.

**Comment by fredericksilva** [comment]
Nossa! 
Amigos, fico muito grato em conhecer vocês, criei uma organização chamada **[BlockchainBr](https://github.com/orgs/blockchainbr/)**. Vamos turbinar esse repo juntos? já que estamos falando sobre futuro!

Joguei a semente!

**Comment by marciomansur** [comment]
SHOOOOW! Bora tentar contribuir o maximo que a gente puder galera, fazer todo mundo conhecer essa tecnologia.

**Comment by rubenmarcus** [comment]
Existe Curso pra isso?

**Comment by rubenmarcus** [comment]
Já viram isso aqui?
https://numer.ai/

https://www.wired.com/2016/12/7500-faceless-coders-paid-bitcoin-built-hedge-funds-brain/
https://www.wired.com/2017/02/ai-hedge-fund-created-new-currency-make-wall-street-work-like-open-source/

**Comment by marciomansur** [comment]
@fredericksilva vamos criar um repo lá para criarmos susgestões para um pontapé inicial!

Tambem podemos colocar la material para aprender sobre mining, desenvolvimento e arquitetura.

**Comment by marciomansur** [comment]
@rubenmarcus Esse curso aqui da Udemy é legal para quem está iniciando. Fala sobre a plataforma e solidity.

https://www.udemy.com/blockchain-developer/

Aqui tenho alguns links úteis:

http://consensys.github.io/developers/articles/101-noob-intro/ <- essa intro é legal demais, para entender a arquitetura do blockchain, como se desenvolve para blockchain e como utilizar aplicações distribuídas.

https://karl.tech/learning-solidity-part-1-deploy-a-contract/ <- esse é para aprender a desenvolver e testar um smart contract (base da logica de aplicações que rodam "on top of Ethereum machine")

http://truffleframework.com <- O melhor framework para se desenvolver dApps. Cria uma abstração para criar, compilar e testar smart contracts, além de facilitar a utilização deles na aplicação. Documentação muito boa, tem dicas de como integrar com plataformas de desenvolvimento e ainda tem um projeto de exemplo (Ethereum pet shop - um pet shop que funciona na blockchain)

**Comment by marciomansur** [comment]
Talvez esse seja um dos overviews mais completos que ja vi até agora. Todos que estejam interessados em estudar para ser blockchain specialists, deem uma lida nesse material.

http://truffleframework.com/tutorials/ethereum-overview

**Comment by rubenmarcus** [comment]
@marciomansur  cara : https://tecnoblog.net/219386/assaltos-ethereum-criptomoeda/

e noticias como essa?

**Comment by kazzkiq** [comment]
@rubenmarcus Na notícia que você passou é dito que o que foi hackeado foram softwares de carteira e/ou exchanges, e não a criptomoeda em si.

Apesar do código das criptomoedas também estar suscetível a eventuais falhas e furos de segurança, isso é algo bem raro de acontecer (pelo menos nas moedas mais populares). O mesmo não é verdade para o código de carteiras e exchanges.

A dificuldade de hackear estas moedas (e os protocolos em que elas estão baseadas) ocorre basicamente por conta da validação do código pela própria comunidade (a maioria delas possui código aberto, muitas inclusive hospedam o código [aqui no GitHub](https://github.com/bitcoin/bitcoin)). Essa "auditoria pública", que pode ser feita por qualquer um, atrai centenas de entusiastas em criptografia e computação, e quando alguma possível falha é encontrada, ela é rapidamente corrigida.

Código aberto + mentes brilhantes procurando e corrigindo vulnerabilidades = 🔐 ❤️ 

*(Quem sabe o TSE não aprenda algo da comunidade open-source sobre como lidar com a urna eletrônica, afinal, esconder o código achando que isso previne fraudes, é em si [uma falha de segurança](https://en.wikipedia.org/wiki/Security_through_obscurity))*

**Comment by jmurowaniecki** [comment]
> (Quem sabe o TSE não aprenda algo da comunidade open-source sobre como lidar com a urna eletrônica, afinal, esconder o código achando que isso previne fraudes, é em si uma falha de segurança)

Não podemos dizer que isso é [exatamente uma verdade](http://www.conjur.com.br/2016-mar-09/tse-promove-ataque-hackers-testar-seguranca-urna), pois em teoria estão trabalhando para aumentar a segurança e transparência do projeto</**ironia**>

Rolou uma notícia recentemente de que custaria cerca de [$2 Bilhões](http://politica.estadao.com.br/noticias/geral,impressao-de-voto-vai-custar-r-2-5-bi-diz-tse,70001900669) para capacitar a impressão dos votos - possivelmente o custo de refatorar toda a aplicação pra uma versão crivelmente auditável :rofl: 

---

Mas então gurizada, qual é a ideia? Vamo dá-lhe ou qualé?

**Comment by fredericksilva** [comment]
@marciomansur vou te adicionar como owner e você cria esse repo de sugestões!

**Comment by fredericksilva** [comment]
@jmurowaniecki Se criarmos uma aplicação para as eleições acredito que consigo levar essa pauta para se transformar em realidade! Tenho aquele capital chamado networking!

**Comment by barraponto** [comment]
@kazzkiq e @rubenmarcus blockchains são vulneráveis e a Ethereum foi hackeada algumas vezes, mais famosamente no [caso do DAO hack](https://en.wikipedia.org/wiki/Ethereum#The_DAO_hard_fork), mas [outras vezes recentemente](https://www.reddit.com/r/ethereum/search?q=attack&restrict_sr=on&sort=relevance&t=all). Blockchains continuam sendo uma boa idéia, só não dá pra creditar todas as falhas aos clientes e exchanges...

**Comment by barraponto** [comment]
Aliás, a Yaso vai dar um curso sobre blockchain para políticas públicas daqui a duas semanas em SP: https://www.eventbrite.com/e/blockchain-para-politicas-publicas-tickets-36555339003

Vale a pena conferir também as sugestões de leitura no final da emenda do curso dela: 
http://yaso.is/Curso%20Blockchain%20PP.pdf

**Comment by aferreira44** [comment]
> blockchains são vulneráveis e a Ethereum foi hackeada algumas vezes, mais famosamente no caso do DAO hack, mas outras vezes recentemente.

@barraponto Discordo de que blockchains são vulneráveis. Os ataques que as redes blockchain sofrem (Bitcoin e Ethereum, por exemplo) fazem as transações ficarem mais caras e/ou demorarem mais para serem confirmadas. Dificilmente acontece um gasto duplo e roubo de tokens, somente com acesso à chave privada.

No caso do hack no "The DAO", por exemplo, foi uma vulnerabilidade de um smart contract escrito no blockchain, não na plataforma do Ethereum em si.

**Comment by fredericksilva** [comment]
> blockchains são vulneráveis e a Ethereum foi hackeada algumas vezes, mais famosamente no caso do DAO hack, mas outras vezes recentemente.

@barraponto  Mestre obrigado por compartilhar o evento e o material, porém também discordo sobre a vulnerabilidade da blockchain. Atualmente estão sim nos projetos de ICO e Exchanges... (Resumindo falhas humanas e não do protocolo..)

**Comment by Renatomsilva** [comment]
Opa Pessoal

Trabalho em uma StartUp Brasileira que usa Blockchain para autenticação de documentos entre outras coisas.

Temos dois Pacotes NPM para manipulação de Wallets no Blockchain do Bitcoin e algumas funcionalidades a mais. Se quiserem dar uma olhada.

Ta ai minha contribuição

Uma delas foi feita para um Cliente e outra continuamos dando continuidade.

https://www.npmjs.com/package/mudamos-libcrypto
https://www.npmjs.com/package/originalmy-libcrypto

**Comment by suissa** [comment]
Nice oneeee @Renatomsilva 

Uma coisa que tenho certeza que o blockchain vai dominar são os cartórios, meu sonho é ver eles acabando!!!!

**Comment by Renatomsilva** [comment]
@suissa  Da uma  olhada originalmy.com , basicamente é esta a ideia

**Comment by suissa** [comment]
@Renatomsilva não estão precisando de dev de Node não? HEhehehhee

Mas na real eu ainda preciso começar criando um blockchainzinho na mão para entender toda a teoria que já sei.

**Comment by Renatomsilva** [comment]
Galera 

Nossa é idéia (OriginalMy)  é ter todos nossas  Libs, App entre outras coisas OpenSource. Aos poucos vamos liberando mais coisas.

Quem quiser aproveitar se cadastrar no Site, estamos dando 5 Vouchers Free na Decred.


@suissa Começa com Ethereum, Truffle e TestRPC pra vc ter uma idéia. A parte de Blockchain do Bitcoin é todo com Libs JS e RPC.

Tinhamos 2 vagas abertas pra NodeJs , mas agora já foram preenchidas , mas acho que pro fim do ano tem mais.

**Comment by suissa** [comment]
@Renatomsilva tentei me logar com o FB e ficou nessa tela meinnn 
![screen shot 2017-08-26 at 6 17 00 pm](https://user-images.githubusercontent.com/75607/29745203-cf2b66d2-8a8a-11e7-9ce6-2149098beda1.png)

**Comment by suissa** [comment]
@Renatomsilva já conheço Etherum há um tempo, ja vi palestra e ja codei uns exemplos, mas nunca algo de verdade.

Já mexi com altcoins há um tempinho:

http://nomadev.com.br/altcoins-o-guia-quase-definitivo/
http://nomadev.com.br/cryptsy-btcdoge/
http://nomadev.com.br/bitcoin-cryptsy-trocando-criptomoedas/
http://nomadev.com.br/ripple-algu%C3%A9m-conhecia-altcoin/

e se escrevi mais não lembro ehehhehe

Mas tb tinha começado um tuto de uma exchange q eu usava q era zica.

**Comment by Renatomsilva** [comment]
@suissa Vou ver o problema do Face , tks pelo feedback

**Comment by suissa** [comment]
@Renatomsilva se vc estiver usando Passport por favor adicione Twitter, Google e Github pa noisss <3

**Comment by Renatomsilva** [comment]
@suissa Estou usando sim . vou Criar a Issue e tentar agilizar isto. Esta semana some um novo método de autenticação irado

**Comment by Renatomsilva** [comment]
Galera 

Mais uma coisa legal com Ethereum 

https://github.com/eddieoz/truffle-mercado-pokemon

**Comment by suissa** [comment]
Isso aqui tb é massa https://blog.consensusx.com/dissecting-the-two-malicious-ethereum-messages-that-cost-30m-but-couldve-cost-100m-155e023a9500

**Comment by suissa** [comment]
@Renatomsilva achei outro aqui ó http://nomadev.com.br/mcxnow-sua-poupan%C3%A7a-virtual/

**Comment by Renatomsilva** [comment]
Irado

On Sat, Aug 26, 2017 at 6:43 PM, Jean Carlo Nascimento <
notifications@github.com> wrote:

> @Renatomsilva <https://github.com/renatomsilva> achei outro aqui ó
> http://nomadev.com.br/mcxnow-sua-poupan%C3%A7a-virtual/
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/frontendbr/forum/issues/624#issuecomment-325163490>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGwbNkG7OORWXCCeWWYSEUGnOcRDuP_yks5scJFrgaJpZM4OgDey>
> .
>



-- 
Renato Silva - Web Developer
renatomsilva@gmail.com

**Comment by fredericksilva** [comment]
Voltei, que lindo isso aqui cara! <3

Tenho alguns pontos que gostaria de ajuda para conectar, comunidade opensource and comunidade cryptocurrencys. Apesar de ser  dois termos que estão no praticamente no mesmo campo de atuação ainda rola uma escassez imensa de developers resolvendo issues de currencys, protocolos gerando assim um grande numero de oportunidades para trabalho e aprendizado. 

Quem topa solucionarmos toda essa escassez? pegando issues resolvendo juntos ou forkando novos repos e criando nossos tokens,contratos,moedas ... (de teste ou não).

vamos nessa?

**Comment by jmlavoier** [comment]
@fredericksilva Cara, desde o começo de 2017 tenho dado uma olhada sobre o assunto e tenho vontade de estudar e quem sabe poder contribuir também. Acredito que isso seria uma ótima experiência.

**Comment by fredericksilva** [comment]
@jmlavoier vem com tudo cara é um mercado muito aquecido!

**Comment by didiraja** [comment]
Passei o final do ano, lendo e discutindo sobre Criptomoedas com meu irmão e fico feliz d++++ de ver esse tópico aqui.

Um comentário levemente off-topic, olhando as exchanges gringas e brasileiras, é facil perceber que elas pecam em vários quesitos. A maioria tem um design fraco, ou tem um limite de criptomoedas oferecidas, isso quando não se limitam á Bitcoin.

Até considerei montar algo com meu irmão mas falta investimento pra isso. Vamos ver se consigo entrar no mercado como dev.

**Comment by renatosramos** [comment]
Ótimo tópico, acredito que em um futuro próximo diversas aplicações já estarão rodando com protocolo blockchain, que vai muito além das criptomoedas em si. Basta ver os players que estão entrando e investindo na tecnologia.

Comecei a estudar blockchain e ethereum recentemente e achei este curso aqui da IBM que acredito contribuir muito pra quem está começando assim como eu: https://www.ibm.com/developerworks/br/cloud/library/cl-ibm-blockchain-101-quick-start-guide-for-developers-bluemix-trs/index.html

**Comment by rsmelo** [comment]
Eu to fazendo ainda esse curso na udemy, ele entrou em promoção novamente agora no inicio do ano. Pra quem quer entender a tecnologia ele é bem bom, to quase terminando ele e achei bem didático https://www.udemy.com/blockchain-developer/learn/v4/overview
